packages/core/src/physics/PhysicsManager.ts

```diff
@@ -0,0 +1,228 @@
+import { HitResult } from "./HitResult";
+import { Ray } from "@oasis-engine/math";
+import { IPhysics, IPhysicsManager } from "@oasis-engine/design";
+import { Collider } from "./Collider";
+import { Layer } from "../Layer";
+import { ColliderShape } from "./shape/ColliderShape";
+
+/**
+ * Filtering flags for scene queries.
+ */
+export enum QueryFlag {
+  STATIC = 1 << 0,
+  DYNAMIC = 1 << 1,
+  ANY_HIT = 1 << 4,
+  NO_BLOCK = 1 << 5
+}
+
+/** A physics manager is a collection of bodies and constraints which can interact. */
+export class PhysicsManager {
+  /** Query flag for raycast */
+  static _queryFlag: QueryFlag = QueryFlag.STATIC | QueryFlag.DYNAMIC;
+  /** @internal */
+  static nativePhysics: IPhysics;
```

Comment:
For `internal`, should add `_`, rename it to `_nativePhysics`.

Reply:
done